### PR TITLE
The Great Big Lavaland Update

### PR DESCRIPTION
**Pull request by MacHac**
This PR does a bunch of stuff, and I have cleverly arranged it into a single PR in order to improve the chance that some of my more controversial features will be merged.  My japes know no equal.
1. FIXES MISSING 96x96 ICONS
   -Dragon swoop icon (you can fight them again, hooray)
   -Some misc clock cult icons (no clue what these do, but TG had them)
2. ELITE HARDSUIT
   elitehardsuitpreview
   -Yes it's an ERT hardsuit recolor, but it's clearly visually distinct from any of the ERT hardsuits.
   -Makes you ash-storm proof and lava proof when worn with the helmet up.
   -Slightly higher base resists than standard explorer suit.
   -Costs 6000 points at the mining vendor.
   -Gives a tiny little bit of incentive to miners to actually do their damn jobs.
3. MEDICAL SUPPLIES
   lavaland medical supplies
   -Single toxin and regular medkit replaced by 3 brute kits and 3 burn kits.
   -Why was there even a toxin kit in there anyway?
   -Miners no longer have to raid medbay at roundstart for actual supplies.
4. SHUTTLE COOLDOWN
   -The mining shuttle now operates on a thirty second cooldown.
   -This prevents idiots from moving the shuttle immediately after it leaves, resulting in people falling down while exiting and being sucked out the open airlock.
5. REQUIRED RUINS
   -It is now possible to designate ruins which will always attempt to spawn at roundstart.
   -This is currently only used for the boss arenas.
6. BOSS ARENAS
   colossuslair
   bubblegumthrone
   drakearena
   -Megafauna are no longer just regular mobs with extremely low spawning chance
   --This means they often generate in places that are nearly impossible to fight them in.
   --This also means that they often would spawn clumped up, making for an impossible fight.
   -Each type of megafauna now has its own arena, which adds to the atmosphere
   --Most of the weapons in these arenas are NODROP, requiring the user to compromise to gain, say, the claymore
   ---Perspicacious viewers will also note the syndicate LMG in the bubblegum arena.  This has a syndicate firing pin, so only nuke ops can use it.  Frankly, if a nuke ops team wants to risk their entire mission to go running to who knows where in lavaland AND fight bubblegum for it, they probably deserve their stupid NODROP LMG.

### CODE REVIEW

**Comment by MacHac** [comment]
GOD DAMN YOU MAPMERGE!  GOD DAMN YOU TO HELL!

**Comment by Super3222** [comment]
Holy madman

Good shit Mac, good shit.

NODROP gear is meh, in my opinion. Sorta weird.

**Comment by Stealthkibbler** [comment]
those corpses in the colossus arena look very tantalizing in the eyes of an ass walker

**Comment by GrayRachnid** [comment]
IMO remove your colossus design and implement /tg/s it's a lot cleaner tbh.
https://github.com/tgstation/tgstation/pull/18398/ This PR for colossus arena.

And remove the nodrop weapons.

**Comment by imtakingabreakdontatme** [comment]
> miners no longer rely on medbay for med supplies

Why even have a department dedicates to healing if people dont need it to heal

Also keep in mind you can replace firing pins in guns as well as far as the lmg goes. Most people probably wont bother anyway though.

Bosses spawning in arenas is probably something I need to do on /tg/ though

File: code/modules/clothing/spacesuits/hardsuit.dm [hunk 1]
```diff
@@ -185,7 +185,56 @@
...
+
+/obj/item/clothing/head/helmet/space/hardsuit/mining/elite/dropped(mob/user)
+	var/mob/living/carbon/human/H = user
+	if(!istype(H))
+		return ..()
+	if(H.wear_suit) //Bunch of precautionary measures in here to protect against admin fuckery causing runtimes.
+		var/obj/item/clothing/suit/space/hardsuit/mining/elite/M = H.wear_suit
+		if(istype(M))
+			if(M.l_int)
+				H.weather_immunities = list("lava") //They already had the lava resist intrinsic when they put on the suit (currently only from ash drake blood)
```
Comment (imtakingabreakdontatme) [inline code/modules/clothing/spacesuits/hardsuit.dm:229]: Shouldnt this be += lava, just incase they are immune to something else?

Comment (imtakingabreakdontatme) [inline code/modules/clothing/spacesuits/hardsuit.dm:229]: Sorry, meant |= "lava"

File: code/modules/clothing/spacesuits/hardsuit.dm [hunk 2]
```diff
@@ -185,7 +185,56 @@
...
+				return
+
+		H.weather_immunities = list()
```
Comment (imtakingabreakdontatme) [inline code/modules/clothing/spacesuits/hardsuit.dm:232]: -= "lava"

**Comment by ShadowDeath6** [comment]
> why even have a department dedicated to healing if miners have their own healing

Well, if miners just steal their shit anyway and dick off, it's negatively impacting medbay anyway since they're losing supplies. Miners can just use the sleepers in survival capsules, anyway. 
The only problem I have is with patches being so good you can just slap one on you and have instant full heal of that damage and then having fleshmend for the next 3 minutes or however long it lasts in their system.

**Comment by Super3222** [comment]
3 bruise and burn kits is an overkill. Especially with the mining medic rework coming up. Added what Shadow said about sleepers, which only requires you to have a helping hand.

There's also bruise kits you can buy from mining vendors for actually their jobs.

**Comment by Super3222** [comment]
3 bruise and burn kits is an overkill. Especially with the mining medic rework coming up. Added what Shadow said about sleepers, which only requires you to have a helping hand.

There's also bruise kits they can buy from mining vendors for actually their jobs.

**Comment by Stealthkibbler** [comment]
Shadows right on the medical part, patches are so good since they let you survive longer in a fight, none of that war of attrition bullshit. You either do a full pullback or you just go ballsdeep in that shit with patches or legion souls. Besides that unless a miner bothers to patrol on the station with his cool new holoparasite buddy, which never happens, then they really dont have much impact on the round outside of minerals and maybe the occasional slain Megafauna.

@Super3222 @ShadowDeath6 At the moment you can buy your own brute kits for 600-700 points, or around that much, if the amount of kits will be lowered than make it possible to also buy poison kits and burn kits from the vendor. 

@KorPhaeron The vendor on its own already eliminates a lot of the duties that medical have on miners, provided that in the mind of a miner, more time spent on miining means more phat loot. Less time spent on mining is time wasted unless its to pickup mod kits/ Better KAs or bluespace equipment which happens to be on the other side of the station for us since we dont map rotate. I dont know how other miners play, but I ALWAYS fill atleast 2 satchels of ores excluding iron and sand in the fist 15 or so minutes since I pretty much NEED that brute kit if I'm gonna be doing any fighting at all.

There's also the fact to consider that we have 5 miners, that are most likely going to get murdered somehow. Patches tend to do a good job because all the fauna usually chases very aggressively not giving you time to patch up and salve yourself (I  believe TG has it so you can instant heal yourself with packs and salves, not sure if you guys still do, but here its instant if you have a buddy doing it, but it takes time to patch yourself up) I think its like, 4 patches per kit, the usage is going to completely depend on world gen, a goliath tendril will potentially cause a miner to use almost all his brute patches, watchers after the update remove the high demand for burn kits, making the almost worthless but raises the need for charcoal from antitox kits drastically. 

Don't get me wrong, people will still use legion souls, they're instant all limb heals that also cure status effects and give you recover from stamina damage which again, the watcher bolamine change is going to make them even more better to use.

My opinion after all that?

Lower burn kit to 1, have 2 poison kits and 2 brutes OR 2 burn kits, 1 poison kit and 3 brutes.
You should also include all the kits in the vendor, possibly consider lowering brute down to around 500 from 600-700. Encouraging miners even more to deliver minerals for their own medical supplies, it pretty much solves the problems that we supposedly have with miners not delivering minerals on its own.

**Comment by imtakingabreakdontatme** [comment]
Yeah I guess my other objection is that having so many kits just lets you facetank things with patch spam, instead of rewarding people for teamwork/having little checkpoints (pod sleepers)

hivelords are still free health anyway

**Comment by imtakingabreakdontatme** [comment]
Just be careful not to sand down all the edges on lavaland. It being diffiult and dangerous is what attracts peoplr to go in the first place. Making it too safe would make it boring.

If you want a comfy "quality of life" you can always hang out on station.

**Comment by Super3222** [comment]
You start yourself up from the bottom in lavaland.
You have a single bruise patch, a bluespace shelter which has a fully operational sleeper and five already-hot donkpockets, and a first-aid kit if you steal it first before your mates do. You basically start using ghetto ways to heal yourself, and if you run out of that you have to resort to tearing off your clothes to use them to craft makeshift bandages/guazes.

Adding classic medical equipment to the mining base is bad IMHO, because then miners won't bother to go medbay and won't bother to mine because they have all the medical equipment they need. Kor takes the words right out of my mouth, what attracts people to go to lavaland is that it's difficult, dangerous, and not everyone is going to make it. Let's not comfort every situation with a nice and soft pillow.

**Comment by Stealthkibbler** [comment]
@Super3222 Miners shouldnt have to go to medbay to get healed, they should have their own ways of doing so, which is already covered with legion souls, the only problem is that you arent always going to find legions.

Given what Kor has said I think it might be better to just remove kits and have 1 burn and brute kits worth of patches laid out on the table, logically a single miner wont take them all because of his limited storage even with the explorers belt. Ephs bandage crafting could do with a bit of time reduction imho, the gauze literally takes a whole fucking ash storm to craft on its own. You're also gonna be making a lot of them due to the fact they only heal one limb.

File: code/modules/clothing/spacesuits/hardsuit.dm (hunk 1 above)
Comment (MacHac) [inline code/modules/clothing/spacesuits/hardsuit.dm:229]: Well, as far as I know, there's currently no way for any human to gain any weather resist except lava.  However, you're right in that I shouldn't be hardcoding it.  I'll fix it next commit.

**Comment by MacHac** [comment]
It's supposed to be difficult and dangerous.  That's why every time I've nerfed something in lavaland, I make something harder to compensate.

I don't think the medic kits are that big of an issue.  There's three of each, and five miners.  This necessitates sharing, or at the very least going without.  The patches are good for semi-serious fuckups, but keep in mind that there's effectively three tiers of danger in lavaland:
- ENVIRONMENTAL (ash storms, lava, chasms): Near instant death in most bad situations with it.
- BOSS (colossus, ash drake, bubblegum, legion): Patches buy time to run away, but not enough to kill it without skill enough to beat it without patches.  Plus, there's usually some warning as to the fact that they are near because of the arenas.
- NON-BOSS (watcher, goliath, legion): Even in large quantities, can be fought solo with stock KA without taking damage (with the exception of watchers grr).

In all of the above fuckups, you're either:
- Likely going to be dying anyway regardless of medical supplies on hand
- Unlikely to be more than inconvenienced by any wounds you take

The patches are there so that solo miners who take a hit or two from a goliath don't have to limp a mile and a half across spider infested station to get basic medical attention.  Teamwork and warm donk pockets are good, but finding a buddy is difficult without metabuddying, and survival capsules are expensive and need to be placed strategically.

I'll consider reducing the amount of availible supplies, but I do think that some conventional medical supplies should be availible without making the trek back to the medbay.  After all, miners are quite possibly the most frequently injured people on the station, barring any traitorous activities.

As for the NODROP gear, I wanted to give players some chance of doing damage to the bosses aside from the standard KA, but I wanted to add some reason to not pick them up and then sprint out of aggro range.

As a side note, there may or may not be one more feature added to this PR before merge, depending on feasability.  I'd like to extend the hitboxes for the boss mobs to their entire size, so that you don't have to unload on their feet to hurt them.

PS I see no reason whatsoever to include a toxin kit.  There's nothing on the asteroid that does toxins damage.  If there is a reason, please enlighten me.

**Comment by Stealthkibbler** [comment]
@MacHac Watchers with bolamine when your watcher changes go through, charcoal in the toxin kit can flush the bolamine out, really useful early on when you dont have stims yet.

**Comment by TheSadFox** [comment]
Shuttle cooldown: 30 seconds is painfully long. Making it something like five seconds and playing an audio cue simmilar to the one that escape shuttle has when its about to depart would be enough to prevent accidental spacings, without wasting even more of miners time.

Medical supplies: Totally against. I even stopped stealing the medkit nowadays. You have so many ways to heal yourself already, the starting patch, donk pockets, bandages, legion souls, implanted legion souls, miners salve and the capsule sleeper if you buddy up with someone. I really don't see a point in adding more meds for miners, other than making lavaland lose its challange. If you want a medkit so badly then go fucking mine 600 points or whatever the cost of a medkit in vendor is around the outpost. And the vendor patches could probably use a nerf too becouse fuck instant full heals during combat. In fact, i wouldn't mind if patches and all the healing chems that are used with them were removed. As much as i love goofchems pyrotechnics and toxins, addition of patches is fucking shit.

Arenas: I'm kinda unsure bout this one honestly. On one hand, if RnD is shit (which they are 9 out of 10 rounds) and doesn't provide you with hyper KA's, you are forced to either lure the boss to the open area to the east of the outpost, which is fairly dangerous, or create the arena yourself, which, without hyper KA, is a painfully slow thing to do. On the other hand, i truly enjoy these WTF moments when i mine a wall just to be set on fire and chased by a drake or shotgunned by colossus and have to run for my life becouse i didn't check the GPS throughly and didn't expect the boss to be hiding behind the wall in that tiny little tunnel.

Hardsuit: Your hardsuit has weaker bullet, energy and laser resistances than the explorer suit.  I'm too lazy to look it up now but can it be upgraded with goliath plates and does it cause any slowdown? Sprite's good, love ERT armor, but i'd honestly prefer if the helmet visor was yellow'ish like the floodlights of a mining hardsuit, personally think it would look better and not trigger my OCD, but yeah, thats just me.

Also LMG in the bubblegums arena might be a bad idea. Stimpack yourself, steal the gun with ease, then ask RD for a pin (why does he fucking starts with a box of pins anyways?) and a plasma cutter, which he will give you most of the time if you give him a story about wanting to fight megafauna/ashwalkers, and you have your sweet 1 hit kill gun. While it probably won't be that bad if used for mining, if a traitor/ling miner gets his hands on it you can be sure that dead chat is going to be extremely salty.

> Given what Kor has said I think it might be better to just remove kits and have 1 burn and brute kits worth of patches laid out on the table, logically a single miner wont take them all because of his limited storage even with the explorers belt.

You start with an internals box which is pretty much a reskinned medkit anyways and an asshole miner could just put all the patches into said box.

**Comment by Stealthkibbler** [comment]
@SeikoTheSeal Chances are that asshole miner will probably die anyway, as mentioned before most the times when you need the patches its usually too late or you barely manage to scrape out of a fight. I've ended up using an entire brute kits worth of patches in one fight with 2 goliaths. Most of the time people will generally patch after a fight regardless of how little the injuries, because of the armor penetration on all the fauna you really dont want to be caught out with a stray 10-20 dmg only for it to make the difference on whether you were gonna be critted or not.

after the watcher rework burn kits will still be useful, especially for ashstorms and it would be nice if we could buy them from the vendor, like I wouldnt even care if it cost 1000 points for one kit, because that shit might save my ass during an ash storm.

**Comment by MacHac** [comment]
In terms of the LMG, like I said, it's cursed.  You get 50 bullets, and then its a permanent and rediculously suspicious arm accessory.  If you're willing to permanently lose the use of one of your arms AND go through all the trouble of avoiding bubblegum long enough to grab it and then trekking back to the station to bug the RD about a pin, all the while dodging security officers who immediately pull lethals on you, then you probably deserve it.  It's way too risky of a strategy for any antag to purposefully attempt, and nonantags can't do much with it anyway.  Plus, one clip will knock only half of a boss's health off.

The elite hardsuit is functionally equivalent to the explorer's suit in terms of goliath plates.  In terms of the bullet resistances, it's not designed to protect against those; It's designed to protect against extreme heat and wildlife.  When comparing it to the explorer suit, keep in mind that its resistances are objectively better than the regular mining hardsuit.  This means, for many station-side traitorous actions, the elite hardsuit is a better choice than the explorer's suit in terms of versatility.

The arenas exist primarily to add some level of atmosphere to the fight, and to give some space and some environmental hazards for the fight.  The whole 'tunnel through a wall right next to one' thing, while pants-shittingly fun, in practice just means death if you found a bubblegum and running away if you found a drake or a colossus.  The arena provides a more regulated, interesting way to fight the monsters, and it ensures that they have some space between each other, and that there's only one of each.

The thirty second cooldown gives 10 seconds for actual travel and 20 for disembarkation.  It can also be used to delay lavaland antags from getting to the station.  20 seconds may seem like a long time, but it becomes more accurate if you have wounded to carry, or multiple people with ore crates, ect.  If we let people move the shuttle while others were unloading and loading, then people get sucked out.

The medkits are purely QOL.  It's a time saver.  If someone edited the map so that mining was next to the medbay, I might see how they would be overkill, but dragging yourself to the other end of the station and waiting around for someone to open medbay for you, only to be ignored by the doctors (who are all gang members at this point), is bad.  It's frustratingly rare to find doctors who both know what they're doing and care enough to do it.

**Comment by imtakingabreakdontatme** [comment]
On our forums we word filter "quality of life" to "powercreep"

**Comment by MacHac** [comment]
@KorPhaeron Point taken.

How about I drop the kits down to one of each, and add burn kits to the mining vendor for the same price as brute kits?

I originally introduced the additional kits to try to prevent miners breaking into medbay storage at roundstart to get the supplies they needed.  That's why I'm hesitant to remove the kits altogether.

**Comment by Stealthkibbler** [comment]
@MacHac shouldnt be a question on adding the kits, just add all kit types to the vendor for the relative price it's pretty much essential for miners to have medical supplies most of the time and even then I've seen miners get killed despite having patches on them.

**Comment by TheSadFox** [comment]
Yeah, adding the kits to the vendor would be a much better way to go than just giving them to miners for free.

**Comment by ghost** [comment]
@MacHac 

claymore is fucking overpowered by the way, it's really fucking retardedly overpowered

you'll probably want to make these items holographic akin to the holodeck rather than nodrop so that they are destroyed when leaving boss arena so that people don't grab their claymore rush out of the arena and go murderboning sec with a retarded powerful sword

**Comment by oisin100** [comment]
Cooonflicts.

Also, We have ash proof hardsuits already, Lets not complicate mining by giving them ANOTHER hardsuit.

**Comment by MayhemSailor** [comment]
I'd like to add an additional request about some sort of audio/visual cue (like the audio for disposals outlet, or ion sparkles) that the shuttle is being requested, either locally or remotely. I find people spaced that didn't know the shuttle was responding to a remote call and are knocked down and sucked out, or recreate an episode of dancing with the stars as they step out now vacant doc.

**Comment by Super3222** [comment]
> Lets not complicate mining by giving them ANOTHER hardsuit.

One hardsuit to protect you from ash storms.
Another one to walk on lava, at an extremely high price which no one will almost ever achieve.

I think its pretty fair.

I'd also love for this PR to be revived. It's been dead for 28+ days

**Comment by ghost** [comment]
@oisin100

THIS HAS BEEN OPEN FOR A MONTH PLES MERGE

**Comment by ShadowDeath6** [comment]
it has merge conflicts adam

**Comment by oisin100** [comment]
> THIS HAS BEEN OPEN FOR A MONTH PLES MERGE

Merge conflicts, Get mac to fix them

**Comment by Super3222** [comment]
@MacHac REEEEEEEEEEEEEEEEEEE

**Comment by Cruix** [comment]
Closed for staleness, re-open if the merge conflicts fix.

**Pull request closed without merging**